lib/http3/common.c

```diff
@@ -1349,10 +1366,17 @@ void h2o_http3_send_goaway_frame(h2o_http3_conn_t *conn, uint64_t stream_or_push
 
 void h2o_http3_send_h3_datagrams(h2o_http3_conn_t *conn, uint64_t flow_id, h2o_iovec_t *datagrams, size_t num_datagrams)
 {
+    if (conn->peer_settings.h3_datagram_rfc) {
+        flow_id = flow_id >> 2; // H3 quarter stream id (https://www.rfc-editor.org/rfc/rfc9297#section-2.1);
+    }
```

Comment:
Rather than doing the quartering here, I think that the implementation would be more uniform between the draft and the rfc version if flow_id simply referred to the quarted stream id in the rfc case.

`handle_input_expect_headers` in `lib/http3/server.c` will do the quartering and nothing else in the server will need to know quartering.  The implementation is keeping around the hash table `conn->datagram_flows`, and as long as that is the case, everything else may as well treat the flow_id as opaque.